tests/test_sensor.py

```diff
@@ -261,6 +261,15 @@ async def test_time_charge_complete_charging(
     assert state.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.TIMESTAMP
     assert state.attributes.get(ATTR_STATE_CLASS) is None
 
+    mock_timetravel = mock_now + timedelta(minutes=2)
```

Comment:
Little out of my element here, I want to draw attention to this to make sure this works for testing. Trying to change the time 2 minutes in the future and make sure that the state has stayed the same.

Reply:
I'm not certain what the issue is. Is the test failing? I haven't mocked timestamps before so I'd just be reading the documentation like you are and reviewing the HA examples.

Reply:
No the test is passing just fine. Maybe @carleeno can review the mock_timetravel. I based it off of their recent PR to shore up time tests. I just want to make sure that the state is actually being evaluated two minutes later by using that ```monkeypatch.setattr...```

Reply:
@InTheDaylight14 the mock looks correct, however you're not asserting anything afterwards so it's not actually testing anything.

Add an `assert state.state == charge_complete_str` at the end and see what happens, this will reveal if your logic is good.

Reply:
for an explanation of what the monkeypatch is doing is it's replacing the `homeassistant.util.dt.utcnow` that HA uses with our mocked time. So when `dt.utcnow` is called from `sensor.py` it will actually get your mocked value.